lib/engines/content_object_store/app/models/content_object_store/result_monad.rb

```diff
@@ -0,0 +1,11 @@
+module ContentObjectStore
+  ResultMonad = Struct.new(:success, :message, :object) do
+    def success?
+      success == true
+    end
+
+    def failure?
```

Comment:
We don't actually end up using this one but I have found it a useful inverse to have when using these.